modules/living_spaces_intranet/living_spaces_intranet.module

```diff
@@ -51,6 +52,18 @@ function living_spaces_intranet_user_login_validate(&$form, FormStateInterface $
   }
 }
 
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function living_spaces_intranet_living_spaces_ban_insert(LivingSpacesBanInterface $entity) {
+  if ('global' == $entity->bundle() && !$entity->get('target_user')->isEmpty()) {
+    /** @var \Drupal\user\UserInterface $target */
+    $target = $entity->get('target_user')->entity;
+    $target->block();
+    $target->save();
+  }
+}
+
```

Comment:
What happens if ban is lifted from user? How it becomes unblocked?

Reply:
implemented delete hook for global bans